src/Templates/NServiceBusWindowsService/ProgramService.cs

```diff
@@ -1,118 +0,0 @@
-﻿using System;
-using System.Diagnostics;
-using System.Linq;
-using System.ServiceProcess;
-using System.Threading.Tasks;
-using NServiceBus;
-using NServiceBus.Logging;
-
-namespace NServiceBusWindowsService
-{
-    class ProgramService : ServiceBase
-    {
-        const string runAsServiceArg = "--run-as-service";
-
-        IEndpointInstance endpoint;
-
-        static ILog logger;
-
-        static ProgramService()
-        {
-            //TODO: optionally choose a custom logging library
-            // https://docs.particular.net/nservicebus/logging/#custom-logging
-            // LogManager.Use<TheLoggingFactory>();
-            logger = LogManager.GetLogger<ProgramService>();
-        }
-
-        public static void Main(string[] args)
-        {
-            using (var service = new ProgramService())
-            {
-                // pass argument at command line to run as a windows service
-                if (args.Contains(runAsServiceArg))
-                {
-                    Run(service);
-                    return;
-                }
-
-                Console.Title = "NServiceBusWindowsService";
-                Console.CancelKeyPress += (sender, e) => { service.OnStop(); };
-                service.OnStart(null);
-                Console.WriteLine("\r\nPress enter key to stop program\r\n");
-                Console.Read();
-                service.OnStop();
-            }
-        }
-
-        protected override void OnStart(string[] args)
-        {
-            AsyncOnStart().GetAwaiter().GetResult();
-        }
-
-        async Task AsyncOnStart()
-        {
-            try
-            {
-                var endpointConfiguration = new EndpointConfiguration("NServiceBusWindowsService");
-
-                //TODO: optionally choose a different serializer
-                // https://docs.particular.net/nservicebus/serialization/
-                endpointConfiguration.UseSerialization<NewtonsoftSerializer>();
-
-                endpointConfiguration.DefineCriticalErrorAction(OnCriticalError);
```

Comment:
This line seems to have disappeared from the new version of template, so `OnCriticalError` will never be called now.

Reply:
Good catch @bording !